packages/swing-store/src/transcriptStore.js

```diff
@@ -667,19 +813,36 @@ export function makeTranscriptStore(
     }
     // 'archival' and 'debug' modes accept all spans
 
+    const doCompression =
+      enableTranscriptRolloverCompression && !metadata.isCurrent;
     const artifactChunks = await makeChunkIterator();
     const inStream = Readable.from(artifactChunks);
     const lineTransform = new BufferLineTransform();
     const lineStream = inStream.pipe(lineTransform).setEncoding('utf8');
+    const spanItems = [];
     let hash = initialHash;
     let pos = startPos;
     for await (const line of lineStream) {
       const item = line.trimEnd();
-      sqlAddItem.run(vatID, item, pos, metadata.incarnation);
+      if (doCompression) {
+        spanItems.push(line);
```

Comment:
We're in an async function. It would be a lot more efficient memory-wise to use the stream version of gzip here, and write each item into the stream directly.